src/daemon/processor/mod.rs

```diff
@@ -309,3 +328,102 @@ fn send_frame(
         Err(_) => true,
     }
 }
+
+// parses percentages and numbers in format of "<coord1>,<coord2>"
+fn parse_coords(raw: String, dimensions: (u32, u32)) -> (u32, u32) {
+    let coords = raw.split(',').map(|s| s.trim()).collect::<Vec<&str>>();
+    let x: &str;
+    let y: &str;
+    if coords.len() != 2 {
+        match coords[0] {
+            "center" => {
+                x = "50%";
+                y = "50%";
+            }
+            "top" => {
+                x = "50%";
+                y = "100%";
+            }
+            "bottom" => {
+                x = "50%";
+                y = "0";
+            }
+            "left" => {
+                x = "0";
+                y = "50%";
+            }
+            "right" => {
+                x = "100%";
+                y = "50%";
+            }
+            "top-left" => {
+                x = "0";
+                y = "100%";
+            }
+            "top-right" => {
+                x = "100%";
+                y = "100%";
+            }
+            "bottom-left" => {
+                x = "0";
+                y = "0";
+            }
+            "bottom-right" => {
+                x = "100%";
+                y = "0";
+            }
+            _ => {
+                debug!(
+                    "Invalid position keyword, using center as fallback: {}",
+                    raw
+                );
+                x = "50%";
+                y = "50%";
+            }
+        }
+    } else {
+        x = coords[0];
+        y = coords[1];
+    }
+    let mut parsed_x: u32;
+    let mut parsed_y: u32;
+
+    //parse x coord
+    match x.parse::<u32>() {
+        Ok(x) => parsed_x = x,
+        Err(_) => {
+            debug!("Invalid x coord, trying to parse as percentage: {}", x);
+            if x.ends_with("%") {
+                let x = x.trim_end_matches("%");
+                match x.parse::<u32>() {
+                    Ok(x) => parsed_x = x,
+                    Err(_) => parsed_x = 0,
+                }
+                parsed_x = parsed_x * dimensions.0 / 100;
+            } else {
+                debug!("Invalid x coord, using 0 as fallback: {}", x);
+                parsed_x = 0;
+            }
+        }
+    }
+
+    //parse y coord
+    match y.parse::<u32>() {
+        Ok(y) => parsed_y = y,
+        Err(_) => {
+            debug!("Invalid y coord, trying to parse as percentage: {}", y);
+            if y.ends_with("%") {
+                let y = y.trim_end_matches("%");
+                match y.parse::<u32>() {
+                    Ok(y) => parsed_y = y,
+                    Err(_) => parsed_y = 0,
+                }
+                parsed_y = parsed_y * dimensions.1 / 100;
+            } else {
+                debug!("Invalid y coord, using 0 as fallback: {}", y);
+                parsed_y = 0;
+            }
+        }
+    }
+    (parsed_x, parsed_y)
+}
```

Comment:
All of this should be done when we parse the **command line arguments**, in `cli.rs`, not here. If we detect something invalid, we immediately exit, tell the user and don't even send anything to the processor.

Here's how you can do that:

First, transform `transition_pos` into a `(f32, f32)`. Each of those is a number between 0 and 1 that represents a percentage. You want to parse these values from the command line. So you will call this parsing function to evaluate the command line arguments. You do this like so:

```rust
#[arg(long, env = "SWWW_TRANSITION_POS", value_parser = parse_coords, default_value = "center")]
pub transition_pos: (f32, f32),
```

And then your `parse_coords` will have to return a `Result<(f32,f32), String>`. All of those `debug` statements at incorrect locations will transform into `Err(format!(msg))`.

After all this, if the parsing succeds we will have the correct numbers to send to the daemon, and we won't have to check for any errors. All we have to do is to transform the percentages into screen coordinates.

Now, there is one problem: this only allows for the percentage approach you took, not the absolute location of the pixels. Personally, I think that's fine; I believe it makes more sense for this option to be a percentage than an absolute value. We can make it more complicated later if needed be.